### ISSUE TITLE
artipacked fixer does not add persist-credentials: false when no upload-artifact is present

### ISSUE THREAD

**Issue by HikaruEgashira** [ultra-supara/sisakulint#411]
## Summary

The `artipacked` rule detects `actions/checkout` without `persist-credentials: false` even when no `upload-artifact` step exists in the job, but the fixer (`-fix on`) only registers `AddAutoFixer` when a dangerous `upload-artifact` is found.

This means `sisakulint -fix on` leaves the `artipacked` warning unresolved for workflows that don't use `upload-artifact`.

## Steps to Reproduce

1. Create a workflow with `actions/checkout` but no `upload-artifact`:

```yaml
name: CI
on: push
permissions:
  contents: read
jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v6
      - run: echo "hello"
```

2. Run `sisakulint` — reports `[Medium] actions/checkout without 'persist-credentials: false'`
3. Run `sisakulint -fix on` — warning is still reported, file is **not** fixed

## Expected Behavior

`sisakulint -fix on` should add `persist-credentials: false` to the checkout step.

## Root Cause

In `pkg/core/artipacked.go`, `AddAutoFixer(NewStepFixer(...))` is only called inside `handleUploadArtifact()` (line 194). In `VisitJobPost()` (line 77-96), the error is reported but the comment explicitly skips the fixer:

```go
// Do not add auto-fixer when there's no dangerous upload-artifact
// because the workflow may legitimately need credentials for git push or similar operations
```

## Suggestion

The design intent (not breaking `git push` workflows) is valid, but the current behavior means the fixer silently skips the majority of `artipacked` warnings. Consider:

- Adding the fixer for checkout steps that have no subsequent `git push` / `git` commands in the same job
- Or adding the fixer unconditionally for `persist-credentials` since `actions/checkout@v6` no longer stores credentials in `.git/config` by default (severity is already `Low` for v6+)
- Or at minimum, emitting a `[skipped-fix]` diagnostic so users know the fix was intentionally not applied

## Environment

- sisakulint version: latest (installed via brew)
- OS: macOS
